### ISSUE TITLE
[CI] RecoveryIT.testRecoveryWithSoftDeletes fails in 7.0

### ISSUE THREAD

**Issue by droberts195** [elastic/elasticsearch#41449]
`RecoveryIT.testRecoveryWithSoftDeletes ` failed in https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.0+default-distro-bwc-tests/228/console

This was a case of 7.0.1 being BWC tested against 7.0.0, and it fails because of a warning about types removal:

```
org.elasticsearch.upgrades.RecoveryIT > testRecoveryWithSoftDeletes FAILED
    org.elasticsearch.client.WarningFailureException: method [DELETE], host [http://[::1]:34397], URI [recover_with_soft_deletes/test/0], status line [HTTP/1.1 200 OK]
    Warnings: [[types removal] Specifying types in document index requests is deprecated, use the /{index}/_doc/{id} endpoint instead.]
    {"_index":"recover_with_soft_deletes","_type":"test","_id":"0","_version":2,"result":"deleted","_shards":{"total":2,"successful":2,"failed":0},"_seq_no":2,"_primary_term":1}
        at __randomizedtesting.SeedInfo.seed([6A1D067DD14D7030:FD6818181BA77D53]:0)
        at org.elasticsearch.client.RestClient.convertResponse(RestClient.java:256)
        at org.elasticsearch.client.RestClient.performRequest(RestClient.java:238)
        at org.elasticsearch.client.RestClient.performRequest(RestClient.java:212)
        at org.elasticsearch.upgrades.RecoveryIT.testRecoveryWithSoftDeletes(RecoveryIT.java:327)
```

This can be reproduced on the 7.0 branch using:

```
./gradlew :qa:rolling-upgrade:v7.0.0#bwcTest -Dtests.seed=6A1D067DD14D7030
```

**Comment by droberts195** [comment]
The same problem occurred in https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.0+default-distro-bwc-tests/227/console

I muted the test on the 7.0 branch in 438824c7716a1b3945a89e7fd54c3a6ea74fcefa

**Comment by droberts195** [comment]
I think it just needs the change of https://github.com/elastic/elasticsearch/commit/83e9d0b#diff-3465237a8aa23442466f36dc73fa8c19L303 backporting to the 7.0 branch.  I opened #41450 for this.

**Comment by ywelsch** [comment]
closed by  #41450
